src/metabase/driver.clj

```diff
@@ -594,3 +594,31 @@
   {:added "0.37.0" :arglists '([driver])}
   dispatch-on-initialized-driver
   :hierarchy #'hierarchy)
+
+(defmulti get-sensitive-fields
+  "Return the set of sensitive fields for the driver"
+  {:added "0.39.0" :arglists '([driver])}
+  dispatch-on-initialized-driver
+  :hierarchy #'hierarchy)
+
+(def ^:const default-sensitive-fields
+  "Set of fields that should always be obfuscated in API responses, as they contain sensitive data."
+  #{:password :pass :tunnel-pass :tunnel-private-key :tunnel-private-key-passphrase
+    :access-token :refresh-token :service-account-json})
+
+(defn default-get-sensitive-fields
+  "Gets all sensitive fields that should be redacted in API responses. This includes everything from the
+   sensitive-fields set, along with any custom properties from connection-properties which are marked with a
+   :type :password."
+  [driver]
+  (let [all-fields      (connection-properties driver)
+        password-fields (filter #(= (get % :type) :password) all-fields)]
+    (into default-sensitive-fields (map (comp keyword :name) password-fields))))
```

Comment:
Is there any foreseeable situation where a driver is going to do something different than this exact behavior? Now that I think about it a bit more, I think this might be better off as a utility function in `metabase.driver.util` or something like that. We should be very careful about introducing new methods to the driver API, we have more than enough already and we don't need to make things more complicated if we don't have a good reason for doing so.

Reply:
Fixed in refactor (now just a single utility function, not a multimethod).